File: data/src/main/java/com/example/data/mapper/IMapper.java
package com.example.data.mapper;

import org.jetbrains.annotations.NotNull;

/**
 * Интерфейс для маппинга данных
 * @param <E>   маппинг в entity
 * @param <D>   маппинг в data
 */
public interface IMapper<E, D> {

    String JSON = "json";
    String ROOM = "room";

    @NotNull
    D mapFromEntity(@NotNull E type);

    @NotNull
    E mapToEntity(@NotNull D type);
}

File: data/src/main/java/com/example/data/mapper/IMapper.kt
package com.example.data.mapper

/**
 * Интерфейс для маппинга данных
 * @param <E>   маппинг в entity
 * @param <D>   маппинг в data
 */
interface IMapper<E, D> {
    companion object {
        const val JSON = "json"
        const val ROOM = "room"
    }

    fun mapFromEntity(type: E): D
    fun mapToEntity(type: D): E
}